realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -322,10 +322,12 @@ public void releaseReference(K object) {
             Integer count = references.get(object);
             if (count == null) {
                 throw new IllegalStateException("Object does not have any references: " + object);
-            } else if (count > 0) {
+            } else if (count > 1) {
                 references.put(object, count - 1);
-            } else {
+            } else if (count == 1){
                 references.remove(object);
+            } else if (count <= 0) {
```

Comment:
Just `else` here might be better here. The current version might be error prone in case of future refactorings.